document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -3674,4 +3749,391 @@ private static void testCountApi(
+    // From collection_data.json, we can verify specific tags appear expected number of times
+    assertTrue(tagCounts.containsKey("hygiene"), "Should contain 'hygiene' tag");
+    assertTrue(tagCounts.containsKey("personal-care"), "Should contain 'personal-care' tag");
+    assertTrue(tagCounts.containsKey("grooming"), "Should contain 'grooming' tag");
+
+    // Verify total count matches expected (each document contributes its tag count)
+    int totalTags = tagCounts.values().stream().mapToInt(Integer::intValue).sum();
+    assertTrue(totalTags > 0, "Total tag count should be greater than 0");
+
+    // Print results for debugging
+    System.out.println("Tag counts from unnest operation:");
+    tagCounts.entrySet().stream()
+        .sorted(Map.Entry.<String, Integer>comparingByValue().reversed())
+        .forEach(entry -> System.out.println(entry.getKey() + ": " + entry.getValue()));
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(PostgresProvider.class)
+  void testNestedPostgresCollectionUnnestTags(String dataStoreName) throws IOException {
+    Datastore datastore = datastoreMap.get(dataStoreName);
+    Collection nestedCollection =
+        datastore.getCollection(COLLECTION_NAME); // Default nested collection
+
+    // Query to unnest tags and group by them to get counts
+    Query unnestQuery =
+        Query.builder()
+            .addSelection(IdentifierExpression.of("tags"))
+            .addSelection(AggregateExpression.of(COUNT, ConstantExpression.of("*")), "count")
+            .addAggregation(IdentifierExpression.of("tags"))
+            .addFromClause(UnnestExpression.of(IdentifierExpression.of("tags"), false))
+            .build();
+
+    CloseableIterator<Document> iterator = nestedCollection.aggregate(unnestQuery);
+
+    // Collect results
+    Map<String, Integer> tagCounts = new HashMap<>();
+    while (iterator.hasNext()) {
+      Document doc = iterator.next();
+      JsonNode json = new ObjectMapper().readTree(doc.toJson());
+      String tag = json.get("tags").asText();
+      int count = json.get("count").asInt();
+      tagCounts.put(tag, count);
+    }
+    iterator.close();
+
+    // Verify we have results
+    assertFalse(tagCounts.isEmpty(), "Should have tag counts from nested collection");
+
+    // Verify some expected tag counts based on our test data
+    // From collection_data.json, we can verify specific tags appear expected number of times
+    assertTrue(tagCounts.containsKey("hygiene"), "Should contain 'hygiene' tag");
+    assertTrue(tagCounts.containsKey("personal-care"), "Should contain 'personal-care' tag");
+    assertTrue(tagCounts.containsKey("grooming"), "Should contain 'grooming' tag");
+
+    // Verify total count matches expected (each document contributes its tag count)
+    int totalTags = tagCounts.values().stream().mapToInt(Integer::intValue).sum();
+    assertTrue(totalTags > 0, "Total tag count should be greater than 0");
```

Comment:
Nit: Perhaps a personal coding style. Messages are generally helpful if it can't be easily inferred from the assertion failure. In all these places, I feel the test failure is good enough to capture the message.

Reply:
@suresh-prakash I personally prefer having clear messages, but would rather stick to what norms we follow today. What do you suggest?

Reply:
Well, not a big deal either ways. I just found it redundant since we can easily infer the failure from the line numbers.